pkg/skaffold/debug/transform.go

```diff
@@ -199,6 +199,51 @@ func transformManifest(obj runtime.Object, retrieveImageConfiguration configurat
 // transformPodSpec attempts to configure a podspec for debugging.
 // Returns true if changed, false otherwise.
 func transformPodSpec(metadata *metav1.ObjectMeta, podSpec *v1.PodSpec, retrieveImageConfiguration configurationRetriever, debugHelpersRegistry string) bool {
+	containers := rewriteContainers(metadata, podSpec, retrieveImageConfiguration, debugHelpersRegistry)
+	timeouts := rewriteProbes(metadata, podSpec) // must rewriteProbes after
+	return containers || timeouts
+}
+
+// rewriteProbes rewrites k8s probes to expand timeouts to 10 minutes to allow debugging local probes.
+func rewriteProbes(metadata *metav1.ObjectMeta, podSpec *v1.PodSpec) bool {
+	annotation, found := metadata.Annotations[DebugConfigAnnotation]
+	if !found {
+		logrus.Warn("no debug config found")
+		return false
+	}
+	var config map[string]ContainerDebugConfiguration
+	if err := json.Unmarshal([]byte(annotation), &config); err != nil {
+		logrus.Warn("error unmarshalling config", err)
+		return false
+	}
+
+	changed := false
+	var minTimeout int32 = 10 * 60 // make it configurable?
```

Comment:
Add seconds to clarify this is in seconds as default duration is Nanosecond
```suggestion
	var minTimeoutSeconds int32 = 10 * 60 // make it configurable?
```

Reply:
Will change to a duration.